yarn-project/end-to-end/src/e2e_inclusion_proofs_contract.test.ts

```diff
@@ -70,6 +70,11 @@ describe('e2e_inclusion_proofs_contract', () => {
         expect(receivedOwner).toEqual(owner.toField());
       }
 
+      // We advance block to be able to prove against block with `noteCreationBlockNumber` --> we have to do this
```

Comment:
I think we could alter to use the current header and "skip" some of the inclusion proofs when we have that you are reading the current block. Like if you are trying to prove something is in the current block number, you can use the header and don't need to find anything in the historical ones, so we could have the if. It increases costs a bit, but seems nice from the PoV of devs?

Would still want to separate just for the case where we do not care about the block_number but just the `historical_header` directly. But guess you can easily cover both in the same code to just have one path even if a bit more expensive 🤔

@benesjan let me know what you think about this, think it fits into #4179